distributed/tests/test_versions.py

```diff
@@ -142,3 +144,31 @@ def test_python_version():
     required = get_versions()["packages"]
     assert "python" in required
     assert required["python"] == ".".join(map(str, sys.version_info))
+
+
+@gen_cluster()
+async def test_python_version_error(s, a, b):
+    s.workers[a.address].versions["packages"]["python"] = "3.5.1"
+
+    with pytest.raises(ImportError) as info:
+        async with Client(s.address, asynchronous=True) as client:
+            pass
```

Comment:
For some reason this `pytest.raises` fails to capture the error on my machine. I'm not sure why.

Reply:
It fails on my machine too. 

How about switching to `error_message` approach?
```python
def test_python_version_error(kwargs_matching):
    kwargs_matching["workers"]["worker-0"]["packages"]["python"] = "3.5.1"
    msg = error_message(**kwargs_matching)
    assert "Python" in msg["error"]
    assert "major" in msg["error"]
```